app/models/machine.rb

```diff
@@ -1,5 +1,17 @@
 class Machine < ApplicationRecord
   validates_presence_of :location
-  
+
   belongs_to :owner
+
+  has_many :machine_snacks
+  has_many :snacks, through: :machine_snacks
+
+  def average_snack_price
+    snacks.average(:price).to_f.round(2)
+  end
```

Comment:
You can do this type of formatting with rounding in the view with number_to_currency. Looks like you're using number_to_precision, and for that you'd definitely want to round yourself, but number_to_currency should round for you and also make sure it goes to two decimal places.